core/src/main/java/io/ddf/etl/TransformationHandler.java

```diff
@@ -12,70 +12,108 @@
 import io.ddf.exception.DDFException;
 import io.ddf.misc.ADDFFunctionalGroupHandler;
 import io.ddf.util.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.*;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 public class TransformationHandler extends ADDFFunctionalGroupHandler implements IHandleTransformations {
 
+  private static Logger LOG = LoggerFactory.getLogger(TransformationHandler.class);
+
   public TransformationHandler(DDF theDDF) {
     super(theDDF);
     // TODO Auto-generated constructor stub
   }
 
-  @Override public DDF transformScaleMinMax() throws DDFException {
-    Summary[] summaryArr = this.getDDF().getSummary();
-    List<Column> columns = this.getDDF().getSchema().getColumns();
+  @Override
+  public DDF transformScaleMinMax() throws DDFException {
+    return this.transformScaleMinMax(new ArrayList<>(), Boolean.FALSE);
+  }
 
+  @Override
+  public DDF transformScaleMinMax(List<String> columns, Boolean inPlace) throws DDFException {
+    Summary[] summaryArr = this.getDDF().getSummary();
     // Compose a transformation query
-    StringBuffer sqlCmdBuffer = new StringBuffer("SELECT ");
+    StringBuilder sqlCmdBuffer = new StringBuilder("SELECT ");
+
+    for (int colIndex = 0; colIndex < this.getDDF().getSchema().getNumColumns(); colIndex++) {
+      Column col = this.getDDF().getSchema().getColumn(colIndex);
 
-    for (int i = 0; i < columns.size(); i++) {
-      Column col = columns.get(i);
-      if (!col.isNumeric() || col.getColumnClass() == ColumnClass.FACTOR) {
-        sqlCmdBuffer.append(col.getName()).append(" ");
+      if (!col.isNumeric() || col.getColumnClass() == ColumnClass.FACTOR
```

Comment:
refactored. Here we're checking each column in DDF to see if it's eligible to be scaled